### PR TITLE
Damage Calculator

### PR DESCRIPTION
**Pull request by czn-PS**
OK, I'm new to all this cool stuff so I know some of this won't be correct but I do know it works! So feedback would be appreciated! (I know I should probably be using Tools.getMove etc but the command fucks up when I do so if you know how to fix that go ahead and comment).

### CODE REVIEW

**Comment by czn-PS** [comment]
Holy shit that's a lot errors.

**Comment by czn-PS** [comment]
I'll do what it says and fix it.

**Comment by Slayer95** [comment]
Are we interested in a calc plugin? O.o

The plan was to embed honko's in the battle gui this whole time...

File: chat-plugins/damage-calculator.js [hunk 1]
```diff
@@ -0,0 +1,443 @@
+global.MOVEDEX = require('../data/moves.js').BattleMovedex;
```
Comment (panpawn) [inline chat-plugins/damage-calculator.js]: These shouldn't be globals.

This also isn't how we add globals.

File: chat-plugins/damage-calculator.js [hunk 2]
```diff
@@ -0,0 +1,443 @@
...
+global.POKEDEX = require('../data/pokedex.js').BattlePokedex;
+global.TYPECHART = require('../data/typechart.js').BattleTypeChart;
+
+global.isolate = function(data) {
+	return JSON.parse(JSON.stringify(data));
+};
+
+exports.commands = {
+'!calc': true,
```
Comment (panpawn) [inline chat-plugins/damage-calculator.js]: indentation

File: chat-plugins/damage-calculator.js (hunk 1 above)
Comment (panpawn) [inline chat-plugins/damage-calculator.js]: We should also put this into strict mode on the top: ``'use strict';``

**Comment by panpawn** [comment]
Also note, the comments I made were just general to help you in general; I personally don't have a preference if this actually should or shouldn't be added to the project.

**Comment by AustinXII** [comment]
This seems like a hassle and also doesn't take into account past generations. This also gets rid of a lot of cool things honko's Calc has (all vs one, etc). Imo this is completely un-needed and adds more work that is already done somewhere else.

**Comment by czn-PS** [comment]
@AustinXII considering I'm doing all the work I don't see how it does cause a hassle. @Slayer95 This isn't permanent it's just till we get Honko's calculator implemented and. @panpawn Thanks will do.

**Comment by AustinXII** [comment]
@czn-PS you still are forgetting about the special features that the damage calc currently has... I know quite a few people would be upset losing one vs all, etc

**Comment by czn-PS** [comment]
what do you mean by "losing" this is just an add on you can still get the normal calc by typing /calc url. :/

**Comment by Zarel** [comment]
Honestly, as long as it links to Honkalculator by typing `/calc` by itself, it's fine, but the main problem here is still low-quality code. I'm going to wait this to at least pass `npm test` before I offer more detailed feedback on code quality.

**Comment by czn-PS** [comment]
@Zarel thanks.

**Comment by panpawn** [comment]
Command name idea: if we're keeping ``/calc`` as the link to Honkalculator, then maybe make this new command just ``/damagecalc``

**Comment by Zarel** [comment]
They can share the same name. It's not a big deal.

**Comment by czn-PS** [comment]
Sooooooo I read you're guys stuff and the code looks really nice now I'm not using globals anymore which is awesome! Which has taught me how most of the Tools work and there really convenient! So yeah looking good! :D

**Comment by czn-PS** [comment]
This isn't the final code this is just to display what I have updated and `npm test` doesn't work on my Cloud 9 so Travis is gonna help me out. :)

**Comment by czn-PS** [comment]
OK so now I all I got to do is do what Travis says and it's done!!!

**Comment by panpawn** [comment]
>I'm not using globals anymore 

You still are :|

**Comment by Zarel** [comment]
Are you going to be updating this? If there are no changes in another 7 days, I'm going to close this. You can open a new pull request when your code gets to the point where it passes `npm test`.

**Comment by czn-PS** [comment]
@Zarel yes I will sorry I have had exams at school so was concentrating on that more but I will be updating sorry again!

**Comment by Lord-Haji** [comment]
embedding Honko's calc to the server will be noice

**Comment by sparkychildcharlie** [comment]
This code isn't originally yours.  It comes from sparkyboTTT (I wrote the code about 1 to 1.5 years ago, hence the low quality)... You remind me of (gitbanned) user psnsVGC who tried to PR the exact same code i gave him in a hastebin to ChuChuBoTTT repository in Jan 2016.

(the globals for Pokedex, Movedex, etc were taken from ``battle/battle.js`` where i made them global for the battling AI, as well as use in kunc, damage calc etc...)

**Pull request closed without merging**